final-project/application.py

```diff
@@ -519,183 +519,98 @@ def getProductJSON():
     product_info[0]['number_of_references'] = number_of_references
+    product_info[0]['reference_ids'] = reference_ids
     product_info[0]['reference_titles'] = reference_titles
     product_info[0]['reference_links'] = reference_links
 
-
-    # Helpful Documentation:
-    # MDN Cross-Origin Sharing Functional Overview
-    # https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
-
     # Return product information in the form of a JSON object.
     return jsonify(product_info[0])
 
-@app.route("/product")
-def renderProductPage():
+@app.route("/product", methods=['POST', 'GET'])
+def render_product_page():
 
-    # Request a product from the user.
-    idOfProductsRequested = request.args.get("id")
+    # Keep the user logged in.
+    user_id = session.get("user_id")
 
-    # Ensure the user asked for a product.
-    if not idOfProductsRequested:
-        return apology("Please let us know which product you are looking for")
-    elif idOfProductsRequested == None:
-        return apology("Please let us know which product you are looking for")
+    # If form was submitted, then it was probably an edit request.
+    # Submit the edits.
+    if request.method == "GET":
 
+        # Request a product from the user.
+        id_of_product_requested = request.args.get("id")
 
-    # Display the product html page
-    return render_template("productJSON.html", idOfProductsRequested=idOfProductsRequested)
+        # Ensure the user asked for a product.
+        if not id_of_product_requested:
+            return apology("Please let us know which product you are looking for")
+        elif id_of_product_requested == None:
+            return apology("Please let us know which product you are looking for")
 
-@app.route("/edit")
-@login_required
-def edit_product():
-    """Edit an individual product"""
+        # Display the product html page
+        return render_template("productJSON.html", id_of_product_requested=id_of_product_requested)
 
-    # Keep the user logged in.
-    user_id = session.get("user_id")
+    if request.method == "POST":
 
-    # TODO:
-    # Ensure we can use request.args & .form in a single function.
-    # If we cannot, then we need to send the product id another way.
+        # PART 1: Save the JSON.
 
+        # If edits were made, then save them.
+        if request.get_json():
+            prod_data = request.get_json()
+        # If no edits were made, then stop.
+        else:
+            return apology("Sorry, we didn't find any edits.")
+
+        # Print the product data key-value pairs (debugger).
+        print('It contains', len(prod_data), prod_data.items())
+        for k, v in prod_data.items():
+            if 'reference' in k:
+                print(k)
+                for key, value in v.items():
+                    print('   key', key, 'has value(', value ,')')
+            else:
+                print('key', k ,'has value(', v ,')')
+        print('')
+        print('')
```

Comment:
What if you just printed out a stringified version of our data? This seems overly-elaborate.